Automattic/woocommerce-payments
# Increase the test coverage and fix failing tests

Fixes https://github.com/Automattic/woocommerce-payments/issues/4907

#### Changes proposed in this Pull Request
* [Fix the if-statement](https://github.com/Automattic/woocommerce-payments/pull/5205/files#diff-ad7970a483b7b3582e959f6236b3c69739d80349cb2236536c7db538726ffe2fR172) that was the result of adding token twice for the non-gateway specific requests + simplify the statement. This fixed subscriptions tests.
* Introduce e2e [tests for Blocks checkout](https://github.com/Automattic/woocommerce-payments/pull/5205/files#diff-b49b9c008428ea968ee3257af76dcab68b13408a25fc2389e206bb3d8ed03114R29) with UPE methods turned on
* [Align the shortcode checkout tests](https://github.com/Automattic/woocommerce-payments/pull/5205/files#diff-0bcfb6ae8e7247b36260ac8358cb9a0e84a992343372f0f329d7d05974186755R26) to be executed with UPE enabled
* [Test adding payment methods on My Account page](https://github.com/Automattic/woocommerce-payments/pull/5205/files#diff-5db1dc5474f9c3e4365bf9d1f5298c30c4f97744a0d10aacca234894dcaf2318R21) with turned on UPE
* Remove a not used and duplicated [shortcode UPE test](https://github.com/Automattic/woocommerce-payments/pull/5205/files#diff-6044e75da9acf8b9aa190af4b6e96a71c959400ced7d96cba537736cc7e64bd6)
* Cover token management and payment method(s) parameters handling areas with unit tests

> 💡 Note: After I added some new logic to the `shopper-myaccount-payment-methods-add-delete-success.spec.js` test, it's now failing because of https://github.com/Automattic/woocommerce-payments/issues/5116. After it will be fixed, the test will pass.

> ❕ Note: After https://github.com/Automattic/woocommerce-payments/issues/4724 is done, only `await merchantWCP.activateUpe();` lines should be deleted as this function enables UPE, which won't be the case anymore. E2E should not be blocked after we remove the `activateUpe()` invocation.









#### Testing instructions
* Run `npm run test:e2e` and confirm all tests, but `shopper-myaccount-payment-methods-add-delete-success.spec.js` (see the note above) are green.

-------------------


**Post merge**
